### DISCUSSION TITLE
My build has been failing

### DISCUSSION THREAD

**Discussion by semilore8** [html2app/discussions#5] [General]
Why has my build been failing ??
 --------  Does it has to do with internet connection or file structures ??

**Comment by yandeu** [comment]
Unfortunately I don't store any information about the build.  Probably file structure. Does the example repository work?

For a successful android debug build , you need at least a `index.html` file inside a `www` directory wrapped in a `.zip` archive.

Example:  
[my-app.zip](https://github.com/html2app/discussions/files/10271832/my-app.zip)

**Comment by semilore8** [comment]
Yes I have figured it out it was file structure..
       I have one more complaint about the navigation button when converted
to app.
 The back navigation button on the device quit the app instead of returning
it to the previous page .

Is there any function or API I can use to control the navigation button??

**Reply by yandeu** [reply]
Maybe it works using `pushState` and `popState`. Try to implement it.

**Reply by yandeu** [reply]
It looks like it needs `@capacitor/app`  
https://stackoverflow.com/a/69084017

**Reply by yandeu** [reply]
I just made a working example using the back button. Have a look!
https://github.com/html2app/plugins-example
